holoviews/plotting/bokeh/raster.py

```diff
@@ -258,13 +258,38 @@ def _get_cmapper_opts(self, low, high, factors, colors):
 
     def _get_colormapper(self, eldim, element, ranges, style, factors=None,
                          colors=None, group=None, name='color_mapper'):
+        indices = None
+        vdims = element.vdims
+        if isinstance(style.get("cmap"), dict):
+            dict_cmap = style["cmap"]
+            missing = [vd.name for vd in vdims if vd.name not in dict_cmap]
+            extra = [k for k in dict_cmap if k not in vdims]
+            if missing:
+                missing_str = "', '".join(sorted(missing))
+                raise ValueError(
+                    "The supplied cmap dictionary must have the same "
+                    f"value dimensions as the element. Missing: '{missing_str}'"
+                )
+            elif extra:
```

Comment:
Thinking about it, the cmap might well be a fixed mapping for categories that simply gets supplied: while missing entries is definitely a problem, I'm not sure warning when you have extra entries is going to be more helpful than annoying. 

For instance, if your element is viewing a portion of the data space where certain categories do not appear, I'm not sure you need to warn about that the key is over-specified for that particular case.

Reply:
Okay I removed it